src/analyses/base.ml

```diff
@@ -2372,6 +2372,7 @@ struct
         | Int n when GobOption.exists (BI.equal BI.zero) (ID.to_int n) -> st
         | Address ret_a ->
           begin match eval_rv (Analyses.ask_of_ctx ctx) gs st id with
+            | Thread a when ValueDomain.Threads.is_top a -> invalidate ~ctx (Analyses.ask_of_ctx ctx) gs st [ret_var]
```

Comment:
In light of #1264, this and the other `invalidate`s for `ThreadJoin` should also be shallow I think. `pthread_join` writes the return value, but doesn't dereference its previous value for writing (it even can't because it's `void*`).